developer_manual/core/integration-tests.rst

```diff
@@ -0,0 +1,167 @@
+
+Create a new file for the context, naming it TaskToTestContext.php.
+
+This is an snippet for creating a new context:
+
+.. code-block::php
+    <?php
+
+    use Behat\Behat\Context\Context;
+    use Behat\Behat\Context\SnippetAcceptingContext;
+
+    require __DIR__ . '/../../vendor/autoload.php';
+
+
+    /**
+     * Example Context.
+     */
+    class ExampleContext implements Context, SnippetAcceptingContext {
+      use Webdav;
+    }
+
+Each scenario underlying code belonging to the new feature to test should be added here.
+
+To add a function to run as a scenario step, do the following:
+
+- Use a @When @Given or @Then statement at the beginning.
+- For parameters you could use either the regular expressions way or the :variable way. But using colons is preferred.
+- Document all the parameters of the function an their expected type.
+- Be careful to write the exact sentence that you will write in the gherkin code. Behat won't parse it properly otherwise.
+
+Example code:
+
+.. code-block::php
+  /**
+   * @When Sending a :method to :url with requesttoken
+   * @param string $method
+   * @param string $url
+   */
+  public function exampleFunction($method, $url) {
+
+
+Add a new feature file to features folder. The name should be task-to-test.feature.
+
+The content of this file should be gherkin code. You can use all the sentences available in the rest of core contexts, just use the apropiate trait in your context. For example "use Webdav;" for using webdav related functions.
+
+Lets show an example of a feature file with scenarios:
+
+..code-block::
+    Feature: provisioning
+      Background:
+        Given using api version "1"
+
+      Scenario: Getting an not existing user
+        Given As an "admin"
+        When sending "GET" to "/cloud/users/test"
+        Then the OCS status code should be "998"
+        And the HTTP status code should be "200"
+
+At the beginning we can see the feature name Feature: provisioning and some variables set in the Background section. This sentence 'Given using api version "1"' corresponds with the use of the v1 of the provisioning API.
+
+A scenario requires three parts, "Given", "When" and "Then" sections. "Given" and "Then" can have several sentences joined together by "And", but "Then" statement should be just one. And this should be the function to test. The other parts are preconditions and postconditions of the test. 
+
+To be able to run your new feature tests you'll have to add a new context to config/behat.yml file.
+In the Contexts section add your new context:
+
+..code-block::
+        contexts:
+              - TaskToTestContext:
+                  baseUrl:  http://localhost:8080/ocs/
+
+After the name add all the variables required for your context. In this example we add just the required baseUrl variable.
+
+For running your new tests just do (assuming that your web user is www-data):
```

Comment:
Yes, please add that detail.
I think some people might assume that you only need to check out from git, maybe run make but not setup OC with "occ maintenance:install" or so.